msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -70,35 +69,42 @@ static ClientApplicationBase()
+        }
+
+        ///  <summary>
+        ///  Constructor of the base application
+        ///  </summary>
+        ///  <param name="clientId">Client ID (also known as <i>Application ID</i>) of the application as registered in the
+        ///  application registration portal (https://aka.ms/msal-net-register-app)</param>
+        ///  <param name="authority">URL of the security token service (STS) from which MSAL.NET will acquire the tokens.
+        /// 
+        ///  Usual authorities endpoints for the Azure public Cloud are:
+        ///  <list type="bullet">
+        ///  <item><description><c>https://login.microsoftonline.com/tenant/</c> where <c>tenant</c> is the tenant ID of the Azure AD tenant
+        ///  or a domain associated with this Azure AD tenant, in order to sign-in users of a specific organization only</description></item>
+        ///  <item><description><c>https://login.microsoftonline.com/common/</c> to sign-in users with any work and school accounts or Microsoft personal account</description></item>
+        ///  <item><description><c>https://login.microsoftonline.com/organizations/</c> to sign-in users with any work and school accounts</description></item>
+        ///  <item><description><c>https://login.microsoftonline.com/consumers/</c> to sign-in users with only personal Microsoft accounts (live)</description></item>
+        ///  </list>
+        ///  Note that this setting needs to be consistent with what is declared in the application registration portal
+        ///  </param>
+        ///  <param name="redirectUri">also named <i>Reply URI</i>, the redirect URI is the URI where the STS will call back the application with the security token. For details see https://aka.ms/msal-net-client-applications</param>
+        ///  <param name="validateAuthority">Boolean telling MSAL.NET if the authority needs to be verified against a list of known authorities.
+        ///  This should be set to <c>false</c> for Azure AD B2C authorities as those are customer specific (a list of known B2C authorities
+        ///  cannot be maintained by MSAL.NET</param>
+        ///  <param name="httpManager"></param>
+        ///  <param name="telemetryManager"></param>
         internal ClientApplicationBase(string clientId, string authority, string redirectUri,
-            bool validateAuthority, IHttpManager httpManager)
+            bool validateAuthority, IHttpManager httpManager, ITelemetryManager telemetryManager)
         {
             HttpManager = httpManager ?? new HttpManager();
             CryptographyManager = PlatformProxyFactory.GetPlatformProxy().CryptographyManager;
             WsTrustWebRequestManager = new WsTrustWebRequestManager(HttpManager);
-            TelemetryManager = new TelemetryManager(Telemetry.GetInstance());  // TODO: pass this in?
+            TelemetryManager = telemetryManager ?? new TelemetryManager(Telemetry.GetInstance());  // TODO: pass this in?
```

Comment:
Is this todo actual?

Reply:
Eventually, we need to get rid of the static from the public api.  But you're right, this TODO can't go away until we do that so I'll remove this.